cmd/crossplane/core/core.go

```diff
@@ -59,7 +69,35 @@ func (c *Command) Run(log logging.Logger) error {
 		return errors.Wrap(err, "Cannot get config")
 	}
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{SyncPeriod: &c.Sync})
+	// approach 1:
+	// abstract away `ctrl.NewManager` call in crossplane-runtime
+	// ----------------------------------------------------------
+	// mgr, err := controller.NewManager(cfg, controller.Options{
+	// 	Options: &ctrl.Options{
+	// 		LeaderElection:          c.LeaderElection,
+	// 		LeaderElectionNamespace: c.Namespace,
+	// 		SyncPeriod:              &c.Sync,
+	// 	},
+	// 	ManagerName:      c.Name,
+	// 	ManagerNamespace: c.Namespace,
+	// })
+
+	// approach 2:
+	// keep calling `ctrl.NewManager` with additional config here
+	// ----------------------------------------------------------
```

Comment:
This is my preferred approach; I feel a little bit of repetition is simpler overall than attempting to provide a helper library for this.